cpp/src/quantiles/tdigest/tdigest_aggregation.cu

```diff
@@ -1032,16 +1090,23 @@ std::unique_ptr<column> merge_tdigests(tdigest_column_view const& tdv,
                                        rmm::cuda_stream_view stream,
                                        rmm::device_async_resource_ref mr)
 {
-  // thrust::merge and thrust::merge_by_key don't provide what we need.  What we would need is an
-  // algorithm like a super-merge that takes two layers of keys: one which identifies the outer
-  // grouping of tdigests, and one which identifies the inner groupings of the tdigests within the
-  // outer groups.
+  // Sort the tdigests by the centroid mean within each group and then pass them to
+  // `compute_tdigests()`. Note that the input has been sorted by the key, but has not by the mean
+  // yet within each group. For sorting by the key, see
+  // `store_result_functor::get_grouped_values()`.
```

Comment:
I think it is more accurate to just say that the individual tdigests have been grouped together by key. Each tdigest has it's own locally consistent (sorted) set of centroids.  These centroids get re-sorted together during the merge step. 

Also, note:  there's definitely a merge conflict here with a recently checked in PR: https://github.com/rapidsai/cudf/pull/16780

Reply:
Thanks, I have updated the doc as suggested.

As for the merge conflict, I have noticed that #16780 has been merged to the `branch-24.10`. I wonder when this change will be picked up and merged to the `branch-24.12`. I can fix the conflict afterwards. Do we have some kind of nightly job that merges new PRs to the new branch for us? Or is it going to be a manual work?

Reply:
Merged the branch and fixed the conflicts.